### PR TITLE
[onert] Introduce MultiModelCompiler

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introduces MultiModelCompiler to separate single and multi model nnpackage compilation.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Draft: #10004

### CODE REVIEW

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 1]
```diff
@@ -106,22 +91,12 @@ std::shared_ptr<CompilerArtifact> Compiler::compile(void)
...
-      throw std::runtime_error{"NYI: Disable compilation for multi model is not supported yet"};
-
-    auto executors = std::make_shared<exec::Executors>();
-
-    _nnpkg->primary_model()->iterate([&](const ir::SubgraphIndex &index, ir::Graph &subg) {
-      executors->emplace(ir::ModelIndex{0}, index, std::make_unique<interp::InterpExecutor>(subg));
-    });
-    return std::make_shared<CompilerArtifact>(executors, nullptr);
-  }
+    throw std::runtime_error{"NYI: Disable compilation for multi model is not supported yet"};
```
Comment (glistening) [inline runtime/onert/core/src/compiler/Compiler.cc:94]: @hseok-oh Is it enough to check `_voptions[0]` only? If not, we may introduce option iteration loop to check `disable_compile` and `he_profiling_mode` in the loop.

Comment (hseok-oh) [reply runtime/onert/core/src/compiler/Compiler.cc]: I'll move option check into for loop.

**Review by glistening — APPROVED** [review]
LGTM.

`Compiler` is a class for a single model compiler (typical test) while `MoltiModelCompiler` is for multiple models (special case).

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**